src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
+		name             string
+		testFn           testFn
+		shouldError      bool
+		expectedErrorMsg string
+		expectedResponse interface{}
+	}
+)
+
+func TestStartAutomatedRecovery(t *testing.T) {
+	successfullStartResponse := RecoveryStartResponse{
+		Code:    200,
+		Message: "Automated recovery started",
+	}
+
+	ctx := context.Background()
+	runTest := func(t *testing.T, test requestTestCase) {
+		baseURL := "http://non-existent-url"
+
+		if test.testFn != nil {
+			server := httptest.NewServer(http.HandlerFunc(test.testFn(t)))
+			defer server.Close()
+			baseURL = server.URL
+		}
+		client, err := NewAdminAPI([]string{baseURL}, BasicCredentials{}, nil)
+		assert.NoError(t, err)
+
+		response, err := client.StartAutomatedRecovery(ctx, ".*")
+
+		if test.shouldError {
+			assert.Error(t, err)
+			// Using assert.Contains instead of assert.Equal as some error messages change depending on the environment.
+			assert.Contains(t, err.Error(), test.expectedErrorMsg)
+			return
+		}
```

Comment:
This is not needed since all of the test cases' `shouldError` are false.

Reply:
same as https://github.com/redpanda-data/redpanda/pull/8836#discussion_r1108322149